array-polyfills/forEach/forEach.js

```diff
@@ -1,6 +1,8 @@
 Array.prototype.myForEach = function (cb) {
   for (let i = 0; i < this.length; i++) {
-    if (typeof this[i] !== "undefined") {
+    if (this[i] === undefined) {
+      cb(undefined, i, this);
+    } else{
```

Comment:
### For the following case:
```
const arr1 = [1, 2, 3, , , 6];
arr1.myForEach(el => console.log(el)); 
```
### Expected output is:
``` 
1 2 3 6
```

### But the output here is
```
1 2 3 undefined undefined 6
```

Reply:
But in Javascript empty values have undefined data type

Reply:
Yes, that's the challenge. We have to differentiate between empty values and undefined values.

Reply:
we cannot identify which are empty undefined or explicitly given undefined , we need to give explicitly null data type to empty values to identify it.

Reply:
`forEach` method does it. So we will need to find a solution to that.